improver/psychrometric_calculations/hail_size.py

```diff
@@ -312,43 +296,33 @@ def extract_pressure_at_268(self, temperature_on_pressure: Cube) -> (CubeList):
 
         data = np.ma.masked_greater(re_ordered_temperature.data, 268.15)
         data = np.ma.masked_invalid(data)
+        shape = re_ordered_temperature.data.shape
 
         if "realization" in coord_order:
-            indices = np.ma.notmasked_edges(data, axis=1)[0][1]
-            pressure = re_ordered_temperature.coord("pressure").points[indices]
-            shape = re_ordered_temperature.data.shape
-
+            axis = 1
             rea, lat, long = (shape[0], shape[2], shape[3])  # ignores pressure axis
-            # identifies if there are columns where the entire column is masked
-            if len(pressure) != rea * lat * long:
-                columns = np.ma.all(data, axis=1).flatten()
-                columns_mask = np.ma.getmask(columns)
-                index = np.where(columns_mask)[0]
-
-                for x in index:
-                    pressure = np.insert(pressure, x, -9999)
-                pressure = np.ma.masked_where(pressure == -9999, pressure)
-
-            pressure = pressure.reshape(rea, lat, long)
-
+            max_length = rea * lat * long
         else:
-            indices = np.ma.notmasked_edges(data, axis=0)[0][1]
-
-            pressure = re_ordered_temperature.coord("pressure").points[indices]
+            axis = 0
+            lat, long = (shape[1], shape[2])  # ignores pressure axis
+            max_length = lat * long
```

Comment:
Or just:
```
shape = [crd.shape[0] for crd in re_ordered_temperature.coords(dim_coords=True) if crd.name() != "realization"]
max_length = np.product(shape)
axis = re_ordered_temperature.coord_dims("pressure")
```

which also simplifies away your later reshaping if statement to just require:
```
pressure = pressure.reshape(shape)
```

Reply:
I've done something similer to this and I also realised that the only function that now required order enforcement is the variable_at_pressure so I've removed unneccessary order enforcement from other functions as well as my order enforcement function as it was no longer a useful function to have.